helixbass/tree-sitter-lint
# First pass at rule options

In this PR:
- first pass at allowing rules to receive "options" as part of their configuration

To test:
Per previous PR testing instructions manually re-build the "custom local binary"
If you add the identifier `foo` somewhere in the code-base and `cargo run` you should see it flagged by both the `replace-foo-with/replace-foo-with-bar` and `replace-foo-with/replace-foo-with-something` rules
If you edit `.tree-sitter-lint.yml` and eg remove the `options` for `replace-foo-with/replace-foo-with-something` or change the `options` value to something that's not parseable as a `usize` it should yell at you
Similarly in the rule tests for rules that take options eg in `src/tests/rules.rs` if you change/remove the expected `options` it should not pass the tests

Based on `plugin`
